src/Services/ConfigurationHandler.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\PDGBundle\Services;
+
+use ApiPlatform\PDGBundle\DependencyInjection\Configuration;
+use Symfony\Component\Config\Definition\Processor;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * Read user configuration on runtime
+ */
+final class ConfigurationHandler
+{
+    private ?array $config = null;
+
+    public function get(string $name, $default = null): mixed
+    {
+        $this->parse();
+
+        // Convert "foo.bar" in "['foo' => ['bar' => ...]]"
+        $config = $this->config;
+        $keys = explode('.', $name);
+        foreach ($keys as $key) {
+            if (array_key_exists($key, $config)) {
+                $config = $config[$key];
+                continue;
+            }
+
+            return $default;
+        }
+
+        return $config;
+    }
+
+    private function parse(): void
+    {
+        // First, load config file from PHP_DOC_CONFIG environment variable
+        $configFile = getenv('PHP_DOC_CONFIG');
+        if ($configFile && !is_file($configFile)) {
+            throw new \RuntimeException(sprintf('Configuration file "%s" does not exist.', $configFile));
+        }
+
+        // If PHP_DOC_CONFIG environment variable is not set, try to load config file from default ordered ones
+        if (!$configFile) {
+            $files = [
+                getcwd().'/pdg.config.yaml',
+                getcwd().'/pdg.config.yml',
+                getcwd().'/pdg.config.dist.yaml',
+                getcwd().'/pdg.config.dist.yml',
```

Comment:
use `getcwd()` in a variable? shouldn't `dist` be in first as it should be used before the normal one?

Reply:
> shouldn't dist be in first as it should be used before the normal one?
The `pdg.config.yaml` file should always be used before the dist one. If both files are present, the `pdg.config.yaml` should be used.